src/AWS.Deploy.CLI/ServerMode/Controllers/DeploymentController.cs

```diff
@@ -439,6 +439,39 @@ public async Task<IActionResult> GetCompatibility(string sessionId)
             return Ok(output);
         }
 
+        /// <summary>
+        /// Start a dry run of the deployment.
+        /// </summary>
+        [HttpPost("session/<sessionId>/dryrun")]
```

Comment:
I looked into that approach but the dry run should be an optional parameter. However, Swagger does not support the optional parameter and ends up generating the RestClient with a required parameter which breaks backwards compatibility and forces the IDE to always provide the dry run flag. That's why I went with the approach of adding a new operation.

Reply:
Ah, and we don't have a request shape for `StartDeployment` so that may be breaking anyway - though if there was a time to do it, we're still in preview.

I'll defer to others who have been involved with the project longer, but still a little wary that we're expanding our public surface area this much to fix a testing issue.

Interested in @normj 's opinion if you get a chance.

Reply:
I don't think you can have an optional resource path parameter. To make it optional you would need to make dryrun be a query string parameter. Can you try that and see if that works with the code generator?

Reply:
Actually based on my later comment I don't think we should try and squeeze this into the StartDeployment and instead have a specific method for returning back the CF template that would be deployed.